### PR TITLE
V2.2.2 prep

### PR DESCRIPTION
**Pull request by tgurbich**
- Modified the annotate_gff.py script (removed dashes when IPR accession is unknown, fixed COG handling, added GO-terms, added e-value filter for eggNOG and InterProScan results)
- Fixed a bug in the readme generation script to correctly detect Archaea presence
- Enabled pre-determined MGYG accessions
- Added a script to post-process GTDB results if there are unknown genomes
- Added fallback to NCBI when getting sample location metadata if ENA returns an error (non GCA genomes)
- Added a switch to FastTree from IQTree if number of species reps is >=2k
- Added accounting for unclassified genomes in website JSONs
- Small bug fixes

### CODE REVIEW

File: modules/gtdbtk.nf [hunk 1]
```diff
@@ -43,7 +43,19 @@ process GTDBTK {
     --extension fna \
     --skip_ani_screen \
     --out_dir gtdbtk_results
-
+    
+    process_gtdb_unknowns.py -i gtdbtk_results -p processed
+    
+    if [ -e gtdbtk_results/classify/processed_gtdbtk.bac120.summary.tsv ]
```
Comment (KateSakharova) [inline modules/gtdbtk.nf:49]: probably it makes sense to rename `gtdbtk_results/classify/gtdbtk.bac120.summary.tsv` to `gtdbtk_results/classify/pre-processed_gtdbtk.bac120.summary.tsv` or similar inside **process_gtdb_unknowns.py**. And then `process_gtdb_unknowns.py` can create files with final names: `gtdbtk_results/classify/gtdbtk.bac120.summary.tsv` and `gtdbtk_results/classify/gtdbtk.ar53.summary.tsv`. 
Movings in nextflow script can cause problems. I guess it is easier to control in python script.

**Review by KateSakharova — APPROVED** [review]
Thank you, looks good. I left one comment

**Review by mberacochea — COMMENTED** [review]
I've added some comments

File: modules/gtdbtk.nf (hunk 1 above)
Comment (tgurbich) [reply modules/gtdbtk.nf:49]: I thought about this but I don't think we need them -  we don't share raw results with users (i.e the contents of this folder) and we can't work with differently formatted taxonomies, they break many things. Reformatting does not fundamentally change the meaning here so I think it's ok.

Comment (mberacochea) [reply modules/gtdbtk.nf:49]: OK, but do we keep a log file with which lines we have changed? That could be useful to trace back errors in the future.

Comment (tgurbich) [reply modules/gtdbtk.nf:49]: Added a log for replacements.

File: bin/deduplicate.py [hunk 2]
```diff
@@ -66,13 +76,23 @@ def main(fastas, outdir):
...
         "--outdir",
         required=True,
         help="The output folder for the deduplicated fasta files and the report tsv",
     )
     args = parser.parse_args()
-
-    main(args.fastas, args.outdir)
+    fastas = []
+    if args.input_folder:
+        fastas = [os.path.join(args.input_folder, i) for i in os.listdir(args.input_folder)]
```
[the remark this answers is not in the document]
Comment (tgurbich) [reply bin/deduplicate.py:95]: Will add this to a future release

**Pull request merged**